livy/src/main/java/org/apache/zeppelin/livy/LivySparkRInterpreter.java

```diff
@@ -43,4 +43,16 @@ public LivySparkRInterpreter(Properties property) {
   public String getSessionKind() {
     return "sparkr";
   }
+
+  @Override
+  protected String extractAppId() throws LivyException {
+    //TODO(zjffdu) depends on SparkR
+    return null;
```

Comment:
Right, previous implementation is only applicable for spark but not for pyspark and sparkr. So I made pyspark and sparkr to both implement `extractAppId` and `extractWebUIAddress`.  return null just means we display null as appId in frontend. Throw exception will cause the SparkR interpreter fails